server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -2853,4 +2852,23 @@ public void test_result_column_type_of_nested_unnest_arg_is_not_an_array() throw
                 .hasDataType(DataTypes.INTEGER)
         );
     }
+
+    /*
+     * https://github.com/crate/crate/issues/16887
+     */
+    @Test
+    public void test_cast_text_array_to_object_array() throws IOException {
```

Comment:
I think this test belongs to CastFunctionTest, there is no need to use analyzer.
There you can also test the actual result of the cast as well. Is there another reason, that you think it needs all this extra infra to test it?

Reply:
Maybe even use a `SymbolsTest` case?

Reply:
No, the plain cast worked before see https://github.com/crate/crate/issues/16887 for details. It only occurs in combination with the type from the table definition as part of the analyzer.